packages/cli/internal/pkg/cli/context/manager_deploy.go

```diff
@@ -5,23 +5,63 @@ import (
 	"path/filepath"
 
 	"github.com/aws/amazon-genomics-cli/internal/pkg/aws/cdk"
-	"github.com/aws/amazon-genomics-cli/internal/pkg/logging"
-	"github.com/rs/zerolog/log"
 )
 
-func (m *Manager) Deploy(contextName string, showProgress bool) error {
-	m.readProjectSpec()
-	m.readConfig()
-	m.readContextSpec(contextName)
+func (m *Manager) Deploy(contexts []string) []ProgressResult {
+	m.readProjectInformation()
+	m.deployAllContexts(contexts)
+	return m.progressResults
+}
+
+func (m *Manager) deployAllContexts(contexts []string) {
+	if m.err != nil {
+		var results []ProgressResult
+		for _, context := range contexts {
+			results = append(results, ProgressResult{Err: m.err, Context: context})
+		}
+
+		m.progressResults = results
+		return
+	}
+
+	progressStreams, contextsWithStreams := m.getStreamsForCdkDeployments(contexts)
+
+	description := fmt.Sprintf("Deploying resources for context(s) %s", contextsWithStreams)
+	m.executeCdkHelper(progressStreams, description)
+}
+
+func (m *Manager) getStreamsForCdkDeployments(contexts []string) ([]cdk.ProgressStream, []string) {
+	var progressStreams []cdk.ProgressStream
+	var contextsWithStreams []string
+	for _, contextName := range contexts {
+		m.readContextSpec(contextName)
+		m.setCdkConfigurationForDeployment(contextName)
+		m.clearCdkContext(contextDir)
+		m.setContextEnv(contextName)
+
+		if m.err == nil {
+			progressStream := m.deployContext(contextName)
+			if progressStream != nil {
+				progressStreams = append(progressStreams, progressStream)
+				contextsWithStreams = append(contextsWithStreams, contextName)
+			}
+		}
+
+		if m.err != nil {
+			m.progressResults = append(m.progressResults, ProgressResult{Context: contextName, Err: m.err})
+		}
```

Comment:
Can we move this into `deployContext`? It's already setting `m.progressResults` on failure so it would be within its scope to also set it when `m.err != nil`